[App Service] BREAKING CHANGE: `az functionapp`: Redact key value output on keys set

kamperiadis opened the pull request:
**Related command**
`az functionapp keys set`
`az functionapp function keys set`

**Description**<!--Mandatory-->
Currently, these two commands return the key value as part of the output. In this PR, we are redacting the key value to prevent accidental security leaks through CI logs. 

**Testing Guide**
<!--Example commands with explanations.-->
1. Create functionapp `az functionapp create`
2. Add function app keys (`az functionapp keys set`) or function keys (`az functionapp function keys set`)
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yanzhudd commented on src/azure-cli/azure/cli/command_modules/appservice/custom.py:
```diff
@@ -5381,15 +5381,21 @@ def update_host_key(cmd, resource_group_name, name, key_type, key_name, key_valu
+                                                                key_type,
+                                                                key_name,
+                                                                key=key_info)
+    logger.warning('Keys have been redacted. Use keys list to view.')
```
It is suggested to specify the list command

kamperiadis replied:
Done

yanzhudd commented on src/azure-cli/azure/cli/command_modules/appservice/custom.py:
```diff
@@ -5434,17 +5440,21 @@ def update_function_key(cmd, resource_group_name, name, function_name, key_name,
+                                                                    function_name,
+                                                                    key_name,
+                                                                    key_info)
+    logger.warning('Keys have been redacted. Use function keys list to view.')
```
It is suggested to specify the list command in the warning message.

kamperiadis replied:
Done

yanzhudd commented:
please rerun the test in live mode and upload the recording file.

yanzhudd commented:
It is suggested to specify the command in the PR title:
[App Service] BREAKING CHANGE: `az functionapp`: Redact key value output on keys set

kamperiadis commented:
> please rerun the test in live mode and upload the recording file.

@yanzhudd I have addressed all your feedback. Can you please help review this PR again and merge if it looks good? Thank you!
